### PR TITLE
Updated with required admin scopes

### PR DESCRIPTION
**Pull request by mauricesharp-okta**
## Latest Preview
09-10-24 14:42 EST
https://preview-5057--reverent-murdock-829d24.netlify.app/docs/guides/terraform-enable-org-access/main/

## Description:
- **What's changed?** 

Added section for new requirements for admin scopes
Updated other information

- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-733720-01-enable-terraform](https://oktainc.atlassian.net/browse/mds/OKTA-704299-Terraform/OKTA-733720-01-enable-terraform)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:80** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md](https://github.com/okta/okta-developer-docs/blob/18c4487afa9193e4b61456c85c614281f934b99c/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fterraform-enable-org-access%2Fmain%2Findex.md) | [82](https://okta.acrolinx.cloud/api/v1/checking/scorecards/ff5835d8-c356-4ccf-9d3a-e1a997f2e72b) | 89  | 67  | 68  | 96  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.0429dba9-0ab4-42fe-83ee-e29f5a8018c8?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 1]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
-If the key isn't in the right format, convert it to the correct format using the OpenSSL command line program:
+Check that the generated private key is in PKCS#1 format, which is the format required by the Okta Terraform provider. In that format, the file that contains the private key begins with `-----BEGIN RSA PRIVATE KEY-----`. If the key isn't in the right format, convert it to the correct format using the OpenSSL command line program:
 
 1. In a terminal, go to the file path where you saved the original private key.
 1. Run OpenSSL to convert the key. One of the following command lines should work depending on your operating system and version of OpenSSL:
-`openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY} -traditional`
-`openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
-`openssl-rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
+   `openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY} -traditional`
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:122]: I think you meant these to be a bulleted list?
```suggestion
   * `openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY} -traditional`
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 2]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
+   `openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:123]:
```suggestion
   * `openssl rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 3]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
+   `openssl-rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:124]:
```suggestion
   * `openssl-rsa -in {ORIGINAL_PRIVATE_KEY} -out {CONVERTED_PRIVATE_KEY}`
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 4]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
-   * `ORIGINAL_PRIVATE_KEY`: The file that contains the key generated earlier.
-   * `CONVERTED_PRIVATE_KEY`: The file that contains the converted key.
-
-> **Note:** The file that contains the converted private key must begin with `-----BEGIN RSA PRIVATE KEY-----`. If not, try step two again.
-
-## Assign admin roles
+   In this command:
+   * `ORIGINAL_PRIVATE_KEY`: Set this to the file path of the original key.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:127]: If you're going to repeat "Set this to" in each item, my preference would be to put the placeholder in the sentence as I think it reads slightly easier...
```suggestion
   * Set `{ORIGINAL_PRIVATE_KEY}` to the file path of the original key.
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 5]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
+   * `CONVERTED_PRIVATE_KEY`: Set this to the file path for the converted key.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:128]:
```suggestion
   * Set `{CONVERTED_PRIVATE_KEY}` to the file path for the converted key.
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 6]
```diff
@@ -135,13 +171,15 @@ Create a Terraform configuration that uses the credentials that you created earl
 
    In the previous code sample, add your values to the following fields:
 
-   * `org_name`: Your Okta org name. For example, `exampleOrgName` from the full domain `https://exampleOrgName.okta.com`.
-   * `base_url`: Your Okta org domain.
-   * `client_id`: The client ID of the service app that you created in an earlier step.
+   * `org_name`: Your Okta org name, which is the first part of your organization's Okta domain before `.okta.com`, `.oktapreview.com`, or `.okta-emea.com`. For example, if your Okta domain is `example-org.oktapreview.com`, the org name is `example-org`. Don't include the `-admin` suffix, which corresponds to your Admin Console URL. To confirm your Okta domain in the Admin Console, click your username in the upper-right corner and look for the text below your email address. An Okta Developer org has an org name with the form `dev-<number>`, such as `dev-000001`.
+   * `base_url`: Your Okta org domain. This is the end part of your domain URL: `.okta.com`, `.oktapreview.com`, or `.okta-emea.com`.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:175]: In the code example above, base_url is set to okta.com, but your text here implies it should have the preceding period - .okta.com 
```suggestion
   * `base_url`: Your Okta org domain. This is the end part of your domain URL: `okta.com`, `oktapreview.com`, or `okta-emea.com`.
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 7]
```diff
@@ -135,13 +171,15 @@ Create a Terraform configuration that uses the credentials that you created earl
...
+   * `client_id`: The client ID of the API service app that you created. In the Admin Console, click **Applications**, then click the name of your API service app. In the app editor, click **General**. Under **Client Credentials**, copy the **Client ID** value.
    * `private_key`: Either the path to the private key file or the private key itself. Okta recommends storing the key in a separate location and using a secrets and encryption management system, such as HashiCorp Vault.
-   * `scopes`: A list of scopes required by the Terraform configuration. This example uses the `okta.groups.manage` scope.
+   * `scopes`: The list of API scopes required by your Terraform configuration. The example below creates a group, which requires the `okta.groups.manage` scope.
```
Comment (danielmaharry-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:178]:
```suggestion
   * `scopes`: The list of scopes required by your Terraform configuration. The following example creates a group, which requires the `okta.groups.manage` scope.
```

**Review by danielmaharry-okta — DISMISSED** [review]
Can you check with Susan or Graham if they're called API scopes anywhere else in the docs? I don't think they are.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 8]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
-In the service app, you specify the Okta objects that Terraform can access using scopes. Every action Terraform makes on an Okta object requires a specific scope. For example, the Terraform configuration in this guide creates a test group in Okta that requires the `okta.groups.manage` scope.
+In the Okta API service app that controls your Terraform integration, you specify the Okta objects that Terraform can access using API scopes. Scopes are like permissions to do a certain action on a type of resource. For example, the Terraform configuration in this article creates a test group in Okta, which requires the `okta.groups.manage` API scope.
+
+One way to determine the required scopes for the service app is to consider which ones an admin needs to perform the same kind of action. Next, find the corresponding scopes in the list of [OAuth Admin Management scopes](/docs/api/oauth2/#okta-admin-management) and add them to your service app. For example, adding a group in the Okta Admin console requires the `groups.manage`. In the table, the corresponding scope is `okta.groups.manage`.
+
+You'll also add any required scopes to your Terraform configuration.
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:37]: I have no idea. I'm only acting as a style updater etc.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 9]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
+
+1. Open the Okta API service app that you created in the previous section.
+1. Click the **Okta API Scopes** tab.
+1. Find the desired scope. For this example, find `okta.groups.manage`.
+1. Click **Grant** for the desired scope.
+1. For some scopes, Okta requests confirmation. To confirm, click **Grant Scope**.
+
+Repeat these steps for any other desired scopes.
+
+> **Important**: When you add other resource types to your Terraform configuration, you must make several changes. Grant the new scopes to the app. Modify your Terraform provider setup to request the new scopes. Depending on your existing setup, you may need to add more admin roles to the app. See [Assign admin roles](#assign-admin-roles).
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:92]: Change to Caution

Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:92]:
```suggestion
> **Caution**: When you add other resource types to your Terraform configuration, you must make several changes. Grant the new scopes to the app. Modify your Terraform provider setup to request the new scopes. Depending on your existing setup, you may need to add more admin roles to the app. See [Assign admin roles](#assign-admin-roles).
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 10]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
 ## Create access credentials
 
-The credentials used are a public/private key pair. Store the public key in the service app, and use the private key later when you create a Terraform configuration.
+Use a public/private key pair to control access to the service app used by Terraform. Store the public key in the API service app, and use the private key in your Terraform configuration. You can use either Okta or an external tool to generate the public/private key pair.
 
-> **Note:** If you use an external tool to generate a key pair, audit the code beforehand to maintain org security.
+> **Important:** Choose an access method based on the level of security required for your service app.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:98]:
```suggestion
> **Caution:** Choose an access method based on the level of security required for your service app.
```

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 11]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
 1. In the Admin Console, open the service app and select **General**.
 1. In the **Client Credentials** section, click **Edit** to change the client authentication method.
 1. Select **Public key / Private key** as the **Client authentication** method.
 1. Click **Add key**.
 1. In the Add a public key dialog, click **Generate new key**.
-1. In the **Private key** section, click **PEM**.
+1. In the **Private key** section, click **PEM** to show the PEM format (not JSON) format of the private key.
 1. Copy the PEM private key and save it as a `.key` file.
+   > **Important:** The private key appears in this dialog only once. Losing the private key requires generating a new key pair.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:109]:
```suggestion
   > **Caution:** The private key appears in this dialog only once. Losing the private key requires generating a new key pair.
```

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:80** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md](https://github.com/okta/okta-developer-docs/blob/a627e5b43368a130855c3f520a672cfa9e143479/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fterraform-enable-org-access%2Fmain%2Findex.md) | [82](https://okta.acrolinx.cloud/api/v1/checking/scorecards/6f61ba89-6537-4452-82df-a0ac2d617dd5) | 89  | 66  | 70  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.935d5594-4e57-4124-9192-77d2e0574364?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 12]
```diff
@@ -1,10 +1,10 @@
 ---
 title: Enable Terraform access for your Okta org
-excerpt: Create an Okta app and credentials that Terraform uses to manage the objects in your org.
+excerpt: Create an Okta application and credentials that Terraform uses to manage the objects in your organization.
 layout: Guides
 ---
 
-Create an Okta app and credentials that Terraform uses to manage the objects in your org.
+Create an Okta application and credentials that Terraform uses to manage the objects in your organization.
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:7]: Fixed

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 13]
```diff
@@ -26,83 +26,119 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
-In the service app, you specify the Okta objects that Terraform can access using scopes. Every action Terraform makes on an Okta object requires a specific scope. For example, the Terraform configuration in this guide creates a test group in Okta that requires the `okta.groups.manage` scope.
+In the Okta API service app that controls your Terraform integration, you specify the Okta objects that Terraform can access using API scopes. Scopes are like permissions to do a certain action on a type of resource. For example, the Terraform configuration in this article creates a test group in Okta, which requires the `okta.groups.manage` API scope.
+
+One way to determine the required scopes for the service app is to consider which ones an admin needs to perform the same kind of action. Next, find the corresponding scopes in the list of [OAuth Admin Management scopes](/docs/api/oauth2/#okta-admin-management) and add them to your service app. For example, adding a group in the Okta Admin console requires the `groups.manage`. In the table, the corresponding scope is `okta.groups.manage`.
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:35]: In next commit

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:80** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md](https://github.com/okta/okta-developer-docs/blob/4a3b44372ad95801350910ae858602880c559517/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fterraform-enable-org-access%2Fmain%2Findex.md) | [85](https://okta.acrolinx.cloud/api/v1/checking/scorecards/dd59f6fc-83b9-4162-98ac-f6a179f1f7aa) | 92  | 68  | 77  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.e1817e93-091a-4da2-b732-ef15368d4d5c?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 14]
```diff
@@ -26,83 +26,117 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
-In the service app, you specify the Okta objects that Terraform can access using scopes. Every action Terraform makes on an Okta object requires a specific scope. For example, the Terraform configuration in this guide creates a test group in Okta that requires the `okta.groups.manage` scope.
+In the Okta API service app that controls your Terraform integration, you specify the Okta objects that Terraform can access using API scopes. Scopes are like permissions to do a certain action on a type of resource. For example, the Terraform configuration in this article creates a test group in Okta, which requires the `okta.groups.manage` API scope.
+
+One way to determine the required scopes for the service app is to determine which actions an admin needs to perform the same kind of action. Next, find the corresponding scopes in the list of [OAuth Admin Management scopes](/docs/api/oauth2/#okta-admin-management) and add them to your service app. For example, adding a group in the Okta Admin console requires the `groups.manage` action. In the table, the corresponding scope is `okta.groups.manage`. Add the same scopes to your Terraform configuration.
```
Comment (grahamsmith-okta) [inline packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:35]: I find this opening sentence a little confusing: "...which actions an admin needs to perform the same kind of action." Could it mean this?
"...which permissions an admin needs to perform the same kind of action."

**Review by grahamsmith-okta — CHANGES_REQUESTED** [review]
Two small comments

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md [hunk 15]
```diff
@@ -26,83 +26,117 @@ Create an Okta app and credentials that Terraform uses to manage the objects in
...
 
 > **Note:** See [Control Terraform access to Okta](/docs/guides/terraform-design-access-security).
 
-## Create an app for Terraform access
+## Create an API service app to manage Terraform access
 
-Create an Okta service app that authorizes Terraform to make changes to your org:
+Create an Okta service app that authorizes Terraform to update your org:
 
 1. In the Admin Console, select **Applications** > **Applications**.
```
[the remark this answers is not in the document]
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:45]: Change made in next commit.

File: packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md (hunk 14 above)
Comment (mauricesharp-okta) [reply packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md:35]: Thanks for catching the issue. You are correct. The changes are in the next commit.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score

A minimum Acrolinx Score of **80** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, for example: Correctness. Your content will be clearer and more consistent.

| Article | **Total score**<br />**Required:80** | Word and phrases<br />(Brand, terms)<br />Preferred: 80 | Correctness<br />(Spelling, grammar)<br />Preferred: 80 | Clarity<br />(Readability)<br />Preferred: 80 | Inclusive language<br />(+ accesibility)<br />Preferred: 80 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/terraform-enable-org-access/main/index.md](https://github.com/okta/okta-developer-docs/blob/54ee7cebaec50a6fb158b247eeebc4ad1b27e07c/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fterraform-enable-org-access%2Fmain%2Findex.md) | [85](https://okta.acrolinx.cloud/api/v1/checking/scorecards/e44288c4-4a85-479e-aec5-edaf7d679ff5) | 92  | 68  | 77  | 100  |

  - Scorecard links, by default, are only good for 7 days. [Link for this commit's Content Analysis Report](https://okta.acrolinx.cloud/api/batch/gen.github.3df6973a-a07f-4351-8cd1-7b79539b3684?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Review by grahamsmith-okta — APPROVED** [review]

**Pull request merged**